docs/index.d.ts

```diff
@@ -55,6 +55,29 @@ export declare interface Tracer extends opentracing.Tracer {
    * Returns a reference to the current scope.
    */
   scope(): Scope;
+
+  /**
+   * Instrument a function by automatically creating a span and activating the
+   * span on the function scope.
+   *
+   * The span will automatically be finished when one of these conditions is
+   * met:
+   *
+   * * The function returns a promise, in which case the span will finish when
+   * the promise is resolved or rejected.
+   * * The function takes a callback as its second parameter, in which case the
+   * span will finish when that callback is called.
+   * * The function doesn't accept a callback and doesn't return a promise, in
+   * which case the span will finish at the end of the function execution.
+   */
+  trace<T>(name: string, fn: (span?: Span, fn?: (error?: Error) => any) => T): T;
+  trace<T>(name: string, options: SpanOptions, fn: (span?: Span, done?: (error?: Error) => string) => T): T;
+
+  /**
+   * Wrap a function to automatically call tracer.tracer() when it's called.
```

Comment:
Reading this I feel like I need to go check what `tracer.tracer()` does. If that's needed, then this is ok. But if people can still get benefit from this without knowing that, maybe it can be phrased otherwise?

Reply:
The idea was to not repeat the documentation since it applies the same rules to the callback, but you're right that it's not ideal since calling `tracer.trace()` is really an implementation detail.

Reply:
Added a more detailed explanation. Also, the rules are not 100% the same so it was indeed better to copy them.

Reply:
IMHO it's fine to repeat here since whoever reads this doc, might not have read the other one. Considering they aren't reading the source code top to bottom.